src/aiod_sdk/endpoints/endpoint.py

```diff
@@ -0,0 +1,134 @@
+import abc
+import requests
+import urllib
+import pandas as pd
+from typing import Literal
+
+API_BASE_URL = "http://localhost/"
+LATEST_VERSION = "v1"
+
+
+class EndpointBase(abc.ABC):
+    api_base_url: str = API_BASE_URL
+    latest_version: str = LATEST_VERSION
+    name: str
+
+
+class Endpoint(EndpointBase, abc.ABC):
+    """
+    Abstract class for endpoints.
+
+    This class provides base functionality for endpoint interactions.
+    """
```

Comment:
The use of classes and inheritance here without any instance variables or methods works, but I wonder if simply using modules would be better. It could look something like this:
`settings.py`
```python
api_base_url = ...
```
`calls.py`
```python
def list(..., asset_type: str) -> ...:
  # basically Endpoint.list, using `asset_type` instead of cls.name,
  # and e.g., settings.api_base_url  instead of the inherited values.
  ...
```
`datasets.py`
```python
from functools import partial
import calls 

list = partial(calls.list, asset_type="datasets")
```

We can use other nice things like `functools.update_wrapper` to copy over metadata. We could simply do a `wrapped_list.__doc__ = list.__doc__.replace("ASSET_TYPE", "asset_type")` to avoid manually copying over docstrings and editing only an asset type. It's clear how the functions are to be used (instead of the classes, which are defined but shouldn't be instantiated). You could also bundle the normally shared calls (that would now be defined in `Endpoint`):
`calls.py`
```python
def wrap_common_calls(asset_type: str) -> Tuple[Callable, ...]:
    list_asset = partial(list, asset_type=asset_type)
    list_asset = update_wrapper(list_asset, list)
    list_asset.__doc__ = list_asset.__doc__.replace("ASSET_TYPE", asset_type).
    # probably wrap above stuff again in a separate function because it would be repeated for every shared function below
    count = partial(count, asset_type=asset_type)
    ...
    return list, count, ...
```
`datasets.py`
```python
list, count, ... = wrap_common_calls(asset_type)
```

This is a redesign that can be done almost entirely internally, so it's not something that has to block the PR. But what would you think of this approach?

Reply:
The one thing that is nice about this approach is that you can use `datasets.list` without the `list` shadowing the built-in. Defining `list` as a module-level variable wouldn't be good (we preferably should use a different name then).

Reply:
Nice, it makes sense. Your suggestion avoids making many copies of the docstrings, so I implemented it.